disnake/onboarding.py

```diff
@@ -97,29 +97,25 @@ class OnboardingPrompt(Hashable):
         If ``False``, the prompt will only appear in community customization.
     """
 
-    __slots__ = (
-        "guild",
-        "id",
-        "title",
-        "options",
-        "single_select",
-        "required",
-        "in_onboarding",
-        "type",
-    )
-
-    def __init__(self, *, guild: Guild, data: OnboardingPromptPayload) -> None:
-        self.guild = guild
-
-        self.id: int = int(data["id"])
-        self.title: str = data["title"]
-        self.options: List[OnboardingPromptOption] = [
-            OnboardingPromptOption(data=option, guild=guild) for option in data["options"]
-        ]
-        self.single_select: bool = data["single_select"]
-        self.required: bool = data["required"]
-        self.in_onboarding: bool = data["in_onboarding"]
-        self.type: OnboardingPromptType = try_enum(OnboardingPromptType, data["type"])
+    __slots__ = ("id", "title", "options", "single_select", "required", "in_onboarding", "type")
+
+    def __init__(
+        self,
+        *,
+        title: str,
+        options: List[OnboardingPromptOption],
+        type: OnboardingPromptType,
+        single_select: bool,
+        required: bool,
+        in_onboarding: bool,
```

Comment:
```suggestion
        single_select: bool = False,
        required: bool = False,
        in_onboarding: bool = True,
```
thought about this again. These are the defaults set by the API, which would make sense to use here as well.
Setting a default for type is a bit tricky, since the API requires you to use type 1 (instead of 0) if there are more than 12 options (which is undocumented) :/

Reply:
> Setting a default for type is a bit tricky, since the API requires you to use type 1 (instead of 0) if there are more than 12 options (which is undocumented) :/

What if we raise an exception if there are more than 12 options and type is `multiple_choice`?

Reply:
That would work, though I'd rather keep the API error here in case this limit ever changes :thinking: 

Probably just mention in `OnboardingPrompt.options` that `type` should be set to `OnboardingPromptType.dropdown` if there are 13 or more options.
If the limit changes at some point (unlikely, but who knows), only the docs would be incorrect, runtime behavior in the library would be unchanged.